flexmeasures/api/common/schemas/sensor_data.py

```diff
@@ -358,15 +358,20 @@ def load_bdf(sensor_data: dict) -> BeliefsDataFrame:
         source = get_or_create_source(current_user)
         num_values = len(sensor_data["values"])
         event_resolution = sensor_data["duration"] / num_values
+        start = sensor_data["start"]
+        sensor = sensor_data["sensor"]
+
+        if sensor.get_attribute("round-event-start"):
+            start = pd.Timestamp(start).round(sensor.get_attribute("round-event-start"))
```

Comment:
Nice place for a walrus operator?

`if (frequency := sensor.get_attribute ...):`